### PR TITLE
Add in threads that only provides an implmentation method

### PR DESCRIPTION
**Pull request by ikelos**
Reworks threads from using an altered generator to something more generic which helps make the dependency easier to see and reason about.

### CODE REVIEW

File: volatility3/framework/plugins/windows/threads.py [hunk 1]
```diff
@@ -18,6 +18,9 @@ class Threads(thrdscan.ThrdScan):
     _required_framework_version = (2, 4, 0)
     _version = (1, 0, 0)
 
+    def __init__(self):
+        self.implementation = self.list_process_threads
```
Comment (atcuno) [inline volatility3/framework/plugins/windows/threads.py:22]: missing call to super(), which should be before the self.implementation line so it doesn't get overwritten

**Comment by atcuno** [comment]
This looks nice. I left one comment just for the missing super(), also would need to send a parameter to generator now: https://github.com/volatilityfoundation/volatility3/blob/5e96327cb0851fe75d1c7a4cb9ef27641bf119f7/volatility3/framework/plugins/windows/thrdscan.py#L107

File: volatility3/framework/plugins/windows/threads.py (hunk 1 above)
Comment (ikelos) [reply volatility3/framework/plugins/windows/threads.py:22]: Good catch, thanks!

**Comment by ikelos** [comment]
Yep, very good catch, thanks!

**Pull request merged**